plugins/linux-lockdown/fu-linux-lockdown-plugin.c

```diff
@@ -90,13 +90,32 @@ fu_linux_lockdown_plugin_add_security_attrs(FuPlugin *plugin, FuSecurityAttrs *a
 {
 	FuLinuxLockdownPlugin *self = FU_LINUX_LOCKDOWN_PLUGIN(plugin);
 	g_autoptr(FwupdSecurityAttr) attr = NULL;
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(GHashTable) cmdline = NULL;
 
 	/* create attr */
 	attr = fu_plugin_security_attr_new(plugin, FWUPD_SECURITY_ATTR_ID_KERNEL_LOCKDOWN);
 	fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ISSUE);
 	fwupd_security_attr_set_result_success(attr, FWUPD_SECURITY_ATTR_RESULT_ENABLED);
 	fu_security_attrs_append(attrs, attr);
 
+	/* we might be able to fix this */
+	cmdline = fu_kernel_get_cmdline(&error_local);
+	if (cmdline == NULL) {
+		g_warning("failed to key kernel cmdline: %s", error_local->message);
+	} else if (fu_kernel_is_cmdline_mutable()) {
+		const gchar *value = g_hash_table_lookup(cmdline, "lockdown");
+		fwupd_security_attr_set_kernel_current_value(attr, value);
+		if (g_strcmp0(value, "confidentiality") != 0) {
+			fwupd_security_attr_set_kernel_target_value(attr,
+								    "lockdown=confidentiality");
+			fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_CAN_FIX);
+		} else {
+			fwupd_security_attr_set_kernel_target_value(attr, "lockdown=integrity");
+			fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_CAN_UNFIX);
```

Comment:
What about if the kernel has been built with `CONFIG_LOCK_DOWN_KERNEL_FORCE_INTEGRITY` or `CONFIG_LOCK_DOWN_KERNEL_FORCE_CONFIDENTIALITY`, these can't really be changed can they?

Reply:
It looks like you can only discover it from `/proc` if the kernel is built with `CONFIG_IKCONFIG`.  But a lot of distros do put it in `/boot/config-$(uname -r)`.  How about at least trying from `/proc/config.gz` followed by `/boot` and show a warning otherwise?

Reply:
I think using /boot is most portable -- and certainly easiest to implement. Added.

Reply:
Both done now in https://github.com/fwupd/fwupd/pull/6206